ProjectController.py

```
@@ -171,6 +172,16 @@ def Generate_embedded_plc_debugger(self):
                 "types": list(set(a.split("_", 1)[0] for a in enums)),
             }
         )
+
         with open(cfile, "w") as f:
             f.write(debug_text)
-        return cfile, debug_text
+
+        # Wrap debugger code around (* comments *)
+        MD5 = hashlib.md5(open(st_file, "rb").read()).hexdigest()
```

Ruff (0.12.2)

Use a context manager for opening files

(SIM115)